tasks/install_shell.sh

```diff
@@ -432,6 +437,19 @@ do_fetch() {
   return 0
 }
 
+do_apt_helper() {
+  info "Trying apt-helper..."
+  run_cmd "/usr/lib/apt/apt-helper download-file '$1' '$2'" 2>$tmp_stderr
+  rc=$?
+
+  if test $rc -eq 0 && test -s "$2" ; then
+    return 0
+  fi
+
+  capture_tmp_stderr "apthelper"
+  return 1
```

Comment:
All of the previous `do_wget/curl/etc` test for failures like

```
if test $rc -ne 0 || test ! -s "$2"; then
  return 1
fi
```
and then `return 0` at the end. Could you do the same here?

Reply:
done, also added 404 test like other methods